coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -812,13 +814,35 @@ func (server *Server) CompleteJob(ctx context.Context, completed *proto.Complete
 			}
 		}
 
+		var completedError sql.NullString
+
+		for _, gitAuthProvider := range jobType.TemplateImport.GitAuthProviders {
+			if !slice.Contains(server.GitAuthProviders, gitAuthProvider) {
+				completedError = sql.NullString{
+					String: fmt.Sprintf("git auth provider %q is not configured", gitAuthProvider),
+					Valid:  true,
+				}
+				break
+			}
+		}
```

Comment:
I'm not sure how we'd do it faster. This would notify the user of this error when a template is being built, which feels appropriate.

Reply:
Oh, I see now, we need to parse the template first. What about the `UpdateJob()`?

Reply:
Well, we actually need the resources to be created first with a `plan`, so we can't actually do it in `UpdateJob` either (unless we parsed the HCL directly, but I think we should be hesitant to do that because it won't respect language primitives like `count`).